### PR TITLE
chore(nimbus): create `PrimaryActionDropDown`

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

Introduces a new `PrimaryActionDropDown` component that combines a primary action button with a dropdown menu. The component operates in two modes: **split button** (primary action + dropdown) when a
`defaultAction` is specified, or **regular button** (single dropdown trigger) when no default action is provided.

### What It Does

The `PrimaryActionDropDown` component lets users quickly access their most common action while providing access to additional related actions through a dropdown menu.

**Split Button Mode** (when `defaultAction` prop is provided):

- Left button executes the primary action immediately
- Right arrow button opens dropdown with all available actions

**Regular Button Mode** (when no `defaultAction`):

- Single button opens dropdown menu when clicked
- Uses provided label text or falls back to "Select an option"

### Key Features

**Smart Error Prevention:**

- Disables dropdown triggers when no actionable menu items exist
- Shows "No actions available" with disabled state for edge cases

**Robust Edge Case Handling:**

- Missing `defaultAction` IDs fallback gracefully
- Empty menus and Menu.Sections are handled properly
- Mixed disabled/enabled states work correctly

**Developer Experience:**

- Type-safe with proper TypeScript guards
- Comprehensive documentation explaining complex logic
- 9 edge case scenarios with automated test coverage

### Usage

This component aligns with recent team discussions revolving around consumer-focused API's. Here, we use the slot API for optional button configuration and expose only the *relevant portions* of the existing Menu API for consumer convenience.

```jsx
// Split button mode
<PrimaryActionDropDown
  defaultAction="save"
  onAction={(id) => handleAction(id)}
  aria-label="Save options"
>
  <Icon slot="icon"><Save /></Icon>
  <Menu.Item id="save">Save Document</Menu.Item>
  <Menu.Item id="export">Export PDF</Menu.Item>
</PrimaryActionDropDown>

// Regular button mode
<PrimaryActionDropDown
  onAction={(id) => handleAction(id)}
  aria-label="Document actions"
>
  <Text slot="label">Document Actions</Text>
  <Menu.Item id="share">Share</Menu.Item>
  <Menu.Item id="download">Download</Menu.Item>
</PrimaryActionDropDown>
```

### CODE REVIEW

File: packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.recipe.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      solid: {
+        dropdownTrigger: {
+          borderLeftColor: "colorPalette.contrast", // White delimiter
+        },
+      },
+      ghost: {
+        dropdownTrigger: {
+          borderLeftColor: "colorPalette.7", // Theme-aware border
+        },
+      },
```
Comment (jaikamat) [inline packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.recipe.tsx:59]: The wireframe warranted slight changes in the delimiters we use

File: packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+          content: item.content,
+          isDisabled: item.isDisabled || false,
+          actionId: defaultAction,
+        };
+      }
+    }
+
+    // If no defaultAction found or no defaultAction specified, show standard fallback
+    return {
+      // TODO: localize this
```
Comment (jaikamat) [inline packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx:222]: We'll have to localize a couple things, I don't expect consumers to have to specify fallbacks. A later time

File: packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+            <PrimaryActionDropDownPrimaryButtonSlot asChild>
+              <Button
+                {...buttonProps}
+                isDisabled={isRegularButtonDisabled}
+                aria-label={ariaLabel}
+              >
+                {iconElement}
+                {buttonContent.content}
+                <KeyboardArrowDown />
+              </Button>
```
Comment (jaikamat) [inline packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx:333]: It was _so nice_ to leverage our own components internally and not have to reinvent the wheel. Unfortunately for me, I discovered this only after spending time reinventing all the wheels

**Comment by jaikamat** [comment]
@FilPob Padding issue should be fixed!
